typescript/src/deposit.ts

```diff
@@ -168,57 +186,106 @@ export async function submitDepositTransaction(
 
 /**
  * Assembles a Bitcoin P2(W)SH deposit transaction.
+ * @dev UTXOs are selected for transaction funding based on their types. UTXOs
+ *      with unsupported types are skipped. The selection process stops once
+ *      the sum of the chosen UTXOs meets the required funding amount.
+ * @param bitcoinNetwork - The target Bitcoin network (mainnet or testnet).
  * @param deposit - Details of the deposit.
- * @param utxos - UTXOs that should be used as transaction inputs.
  * @param depositorPrivateKey - Bitcoin private key of the depositor.
  * @param witness - If true, a witness (P2WSH) transaction will be created.
  *        Otherwise, a legacy P2SH transaction will be made.
+ * @param utxos - UTXOs that should be used as transaction inputs.
+ * @param fee - Transaction fee to be subtracted from the sum of the UTXOs'
+ *        values.
  * @returns The outcome consisting of:
  *          - the deposit transaction hash,
  *          - the deposit UTXO produced by this transaction.
  *          - the deposit transaction in the raw format
+ * @throws {Error} When the sum of the selected UTXOs is insufficient to cover
+ *        the deposit amount and transaction fee.
  */
 export async function assembleDepositTransaction(
+  bitcoinNetwork: BitcoinNetwork,
   deposit: Deposit,
-  utxos: (UnspentTransactionOutput & RawTransaction)[],
   depositorPrivateKey: string,
-  witness: boolean
+  witness: boolean,
+  utxos: (UnspentTransactionOutput & RawTransaction)[],
+  fee: BigNumber
 ): Promise<{
   transactionHash: TransactionHash
   depositUtxo: UnspentTransactionOutput
   rawTransaction: RawTransaction
 }> {
-  const depositorKeyRing = createKeyRing(depositorPrivateKey)
-  const depositorAddress = depositorKeyRing.getAddress("string")
-
-  const inputCoins = utxos.map((utxo) =>
-    bcoin.Coin.fromTX(
-      bcoin.MTX.fromRaw(utxo.transactionHex, "hex"),
-      utxo.outputIndex,
-      -1
-    )
+  const network = toBitcoinJsLibNetwork(bitcoinNetwork)
+  // eslint-disable-next-line new-cap
+  const depositorKeyPair = ECPairFactory(tinysecp).fromWIF(
+    depositorPrivateKey,
+    network
   )
 
-  const transaction = new bcoin.MTX()
+  const psbt = new Psbt({ network })
+  psbt.setVersion(1)
```

Comment:
This is always version 1, regardless of the inputs and outputs time right?

Reply:
Yes, for simple transaction version `1` is enough regardless of the input type and that's what was used in the original `bcoin` lib.